src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenFunctionPointersTests.cs

```diff
@@ -3194,6 +3226,22 @@ unsafe static void Main()
                 //         delegate*<C, void> ptr = &IHelpers.M;
                 Diagnostic(ErrorCode.ERR_AmbigCall, "IHelpers.M").WithArguments("IHelpers.M(I1)", "IHelpers.M(I2)").WithLocation(13, 35)
             );
+
+            var syntaxTree = comp.SyntaxTrees[0];
+            var model = comp.GetSemanticModel(syntaxTree);
+
+            var addressOf = syntaxTree.GetRoot().DescendantNodes().OfType<PrefixUnaryExpressionSyntax>().Single();
+
+            string[] expectedSymbolCandidates = new[] { "void IHelpers.M(I1 i1)", "void IHelpers.M(I2 i2)" };
+            FunctionPointerUtilities.VerifyFunctionPointerSemanticInfo(model, addressOf,
+                expectedSyntax: "&IHelpers.M",
+                expectedType: null,
+                expectedConvertedType: "delegate*<C, System.Void>",
+                expectedCandidateReason: CandidateReason.OverloadResolutionFailure,
+                expectedSymbolCandidates: expectedSymbolCandidates);
+
+            AssertEx.Equal(expectedSymbolCandidates, model.GetMemberGroup(addressOf).Select(m => m.ToTestDisplayString(includeNonNullable: false)));
+            AssertEx.Equal(expectedSymbolCandidates, model.GetMemberGroup(addressOf.Operand).Select(m => m.ToTestDisplayString(includeNonNullable: false)));
         }
```

Comment:
>} [](start = 8, length = 1)

Is it interesting to test IOperation for this scenario, or is it covered elsewhere?

Reply:
Same comment: I didn't consider it particularly different from the failed addressof versions in `FunctionPointerOperations.cs`. I can add verification here as well if you'd prefer though.

---
In reply to: [430770526](https://github.com/dotnet/roslyn/pull/44436#discussion_r430770526) [](ancestors = 430770526)